src/Microsoft.AspNetCore.Razor.Language/AssemblyExtension.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    internal class AssemblyExtension : RazorExtension
+    {
+        public AssemblyExtension(string extensionName, Assembly assembly)
+        {
+            if (extensionName == null)
+            {
+                throw new ArgumentNullException(nameof(extensionName));
+            }
+
+            ExtensionName = extensionName;
+            Assembly = assembly;
+        }
+
+        public override string ExtensionName { get; }
+
+        public Assembly Assembly { get; }
+
+        public void AddExtension(RazorProjectEngineBuilder builder)
```

Comment:
I might change how this works. For instance I kinda want to put this logic inside the `RazorProjectEngine.Create` method, thoughts?

Reply:
I agree with this feedback. I'll give this a try and see if it's small enough to cram into this PR.